tcms/telemetry/static/telemetry/js/management/performance.js

```diff
@@ -0,0 +1,142 @@
+// TODO: account for more users
+const colors = [
+  'blue',
+  'red',
+  'gold',
+  'orange',
+  'green',
+  'cyan',
+  'purple',
+  'black',
+  'lightBlue',
+  'lightGreen'
+]
+
+$(document).ready(() => {
+  $('.selectpicker').selectpicker()
+  $('[data-toggle="tooltip"]').tooltip()
+
+  loadInitialProduct(reloadCharts)
+
+  $('#id_after').on('dp.change', reloadCharts)
+  $('#id_before').on('dp.change', reloadCharts)
+  document.getElementById('id_test_plan').onchange = reloadCharts
+  document.getElementById('id_product').onchange = () => {
+    updateTestPlanSelectFromProduct(reloadCharts)
+  }
+})
+
+function reloadCharts () {
+  const query = {}
+
+  const testPlanIds = $('#id_test_plan').val()
+  const productIds = $('#id_product').val()
+
+  if (testPlanIds.length) {
+    query.plan__in = testPlanIds
+  } else if (productIds.length) {
+    query.category__product_id__in = productIds
+  }
+
+  const dateBefore = $('#id_before')
+  if (dateBefore.val()) {
+    query.create_date__lte = dateBefore.data('DateTimePicker').date().format('YYYY-MM-DD 23:59:59')
+  }
+
+  const dateAfter = $('#id_after')
+  if (dateAfter.val()) {
+    query.create_date__gte = dateAfter.data('DateTimePicker').date().format('YYYY-MM-DD 00:00:00')
+  }
+
+  jsonRPC('Management.performance', query, result => {
+    console.log(result)
+
+    // the actual result is in the same format, only it can be much bigger
+    // and the chart may break
```

Comment:
We need to have an understanding of why & how this chart breaks. If you are seeing breakage locally in devel, that's not a terribly large dataset. 

Both Status matrix & Execution trends pages seem to have a larger dataset, at least in production and they seem to load just fine. On tcms.kiwitcms.org Execution trends takes a few seconds but it contains info from 5564 executions across around 1000 test runs grouped by statuses.

Reply:
I meant this comment for here - https://github.com/kiwitcms/Kiwi/pull/2490#discussion_r711600817